### ISSUE TITLE
Gracefully exit when no TIR is found

### ISSUE THREAD

**Issue by GallVp** [lutianyu2001/TIR-Learner#2]
Hi @lutianyu2001 

Thank you for your amazing work. Would it be possible to update the tool so that when TIRs are not found, the tool exists gracefully with a message saying no TIR was found. Currently, it produces the following error:

```bash
Doing pre-scan for genome file...
Genome file scan finished! Time elapsed: 0.03697 s.
  File name: genome.fasta
  File size: 29.6113KiB
  Number of sequences: 1
  Number of short sequences: 0
  Percentage of short sequences: 0.0 %
  Average sequence length: 29 k
########################################################## Module 4 Begin #########################################################
Module 4, Step 1: Run TIRvish to find inverted repeats
  Step 1/3: Processing FASTA files
  Step 2/3: Executing TIRvish with python multiprocess
  Step 3/3: Getting TIRvish result
Traceback (most recent call last):
  File "/usr/local/share/TIR-Learner3/TIR-Learner.py", line 95, in <module>
    TIRLearner_instance = TIRLearner(genome_file, genome_name, species, TIR_length,
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/share/TIR-Learner3/bin/main.py", line 91, in __init__
    self.execute()
  File "/usr/local/share/TIR-Learner3/bin/main.py", line 131, in execute
    self.execute_M4()
  File "/usr/local/share/TIR-Learner3/bin/main.py", line 584, in execute_M4
    self["TIRvish"] = run_TIRvish.execute(self)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/share/TIR-Learner3/bin/run_TIRvish.py", line 241, in execute
    return get_fasta_pieces_SeqIO(genome_file, df, cpu_cores, flag_verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/share/TIR-Learner3/bin/get_fasta_sequence.py", line 63, in get_fasta_pieces_SeqIO
    return pd.concat(df_with_seq_list).sort_index()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/reshape/concat.py", line 382, in concat
    op = _Concatenator(
         ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/reshape/concat.py", line 445, in __init__
    objs, keys = self._clean_keys_and_objs(objs, keys)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pandas/core/reshape/concat.py", line 541, in _clean_keys_and_objs
    raise ValueError("All objects passed were None")
ValueError: All objects passed were None
```

I used this test file: https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/sarscov2/genome/genome.fasta

**Comment by lutianyu2001** [comment]
Thank you very much for your report! This problem has been fixed in 7d3b6e2be4b07ee1065c26ed522a45a05e9e1bdf. A new pre-release [v3.0.4-beta](https://github.com/lutianyu2001/TIR-Learner/releases/tag/v3.0.4-beta) has also been published. 😉

**Comment by GallVp** [comment]
> Thank you very much for your report! This problem has been fixed in [7d3b6e2](https://github.com/lutianyu2001/TIR-Learner/commit/7d3b6e2be4b07ee1065c26ed522a45a05e9e1bdf). A new pre-release [v3.0.4-beta](https://github.com/lutianyu2001/TIR-Learner/releases/tag/v3.0.4-beta) has also been published. 😉

That's amazing. Thank you @lutianyu2001
